greenplumpython/func.py

```diff
@@ -26,11 +30,54 @@ def to_table(self) -> Table:
 
 
 def function(name: str, db: Database) -> Callable[..., FunctionCall]:
-    def make_callable(*args: Expr, as_name: Optional[str] = None) -> FunctionCall:
+    def make_function_call(*args: Expr, as_name: Optional[str] = None) -> FunctionCall:
         return FunctionCall(name, db, args, as_name=as_name)
 
-    return make_callable
+    return make_function_call
 
 
-def create_function():
-    pass
+def create_function(
+    db: Database,
```

Comment:
I am a bit worried about the db param here.
By specifying the `db` in the `create_function()`, the user has to have a `db` instance before declaring the UDF. And if user has some common UDFs across different database, do they have to start a new greenplumpython process? If that is the case, our `Database` class should just be a singleton and we don't need `db` for each function.

I am thinking about delaying the UDF declaration until the execution when we know which db to use.

I can see it will be some amount of work, we can discuss and implement it in other PRs if needed.

Reply:
I seriously think the UDF creation should be delayed until it is needed. We CAN NOT ask the user to have a valid gpdb connection when she/he is writing and testing the UDF body in its local python env (not the gpdb plpython).

Reply:
Changed.